lib/script/taproot.js

```diff
@@ -0,0 +1,92 @@
+/*!
+* taproot.js - taproot interpreter for bcoin
+* https://github.com/bcoin-org/bcoin
+*/
+
+'use strict';
+
+const bio = require('bufio');
+const common = require('./common');
+const { strcmp } = require('../utils/util');
+
+const {
+  TapLeafHash,
+  TapBranchHash
+} = require('../utils/taggedhash');
+
+/**
+ * TapLeaf
+ */
+
+class TapLeaf {
+  /**
+   * Create a leaf in taproot tree.
+   * @constructor
+   * @param {Script} script
+   * @param {Number} leafVersion
+   * @param {String} name
+   */
+
+  constructor(script, leafVersion=common.LEAF_VERSION_TAPSCRIPT, name='') {
+    this.script = script;
+    this.leafVersion = leafVersion;
+    this.name = name;
+  }
+}
+
+const getLeafHash = (script, leafVersion) => {
+  const size = script.getVarSize();
+  const bw = bio.write(size + 1);
+  bw.writeU8(leafVersion);
+  bw.writeVarBytes(script.raw);
+  const data = bw.render();
+  return TapLeafHash.digest(data);
+};
```

Comment:
can this be a method of `TapScript`? then you could just use `this.script` and `this.leafVersion`